plugins/ipam/host-local/backend/disk/lock_windows.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package disk
+
+import (
+	"github.com/alexflint/go-filemutex"
+	"os"
+	"path"
+)
+
+// FileLock wraps os.File to be used as a lock using flock
+type FileLock struct {
+	f *filemutex.FileMutex
+}
+
+// NewFileLock opens file/dir at path and returns unlocked FileLock object
+func NewFileLock(lockPath string) (*FileLock, error) {
+	fi, err := os.Stat(lockPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if fi.IsDir() {
+		lockPath = path.Join(lockPath, "lock")
+	}
+
+	f, err := filemutex.New(lockPath)
```

Comment:
this code seems identical to the POSIX version, except it makes use of this nice cross-platform adapter.  Could we make this file the one true `lock.go` and remove the POSIX-only implementation entirely?

Reply:
Done